src/components/pages/devices/flyouts/deviceNew/deviceNew.js

```diff
@@ -159,15 +162,23 @@ export class DeviceNew extends LinkedComponent {
     this.countLink = this.formDataLink.forkTo('count')
       .reject(nonInteger)
       .map(stringToInt)
-      .check(Validator.notEmpty, 'Number of devices is required.')
-      .check(num => num > 0, 'Number of devices must be greater than zero.');
-
-    this.deviceIdLink = this.formDataLink.forkTo('deviceId');
+      .check(Validator.notEmpty, () => this.props.t('devices.flyouts.new.validation.required'))
+      .check(num => num > 0, () => this.props.t('devices.flyouts.new.validation.greaterThanZero'));
 
     this.isGenerateIdLink = this.formDataLink.forkTo('isGenerateId')
       .map(stringToBoolean);
 
-    this.deviceModelLink = this.formDataLink.forkTo('deviceModel');
+    this.deviceIdLink = this.formDataLink.forkTo('deviceId')
+      .check(
+        devId => (!this.deviceTypeLink.value && !this.isGenerateIdLink.value ? Validator.notEmpty(devId) : true),
+        () => this.props.t('devices.flyouts.new.validation.required')
+      );
+
+    this.deviceModelLink = this.formDataLink.forkTo('deviceModel')
+      .check(
+        devModel => (this.deviceTypeLink.value ? Validator.notEmpty(devModel) : true),
```

Comment:
No real need to wrap these ternary operator with `()` unless there are several conditions that need to be clarified.